core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -85,13 +86,36 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     */
   private var cachedPendingLaunch = CachedPendingLaunch(Set.empty)
 
+  def validateEvent(msg: SpecEvent): Seq[PodInvalid] = msg match {
+    case SpecsSnapshot(podSpecs, _) =>
+      podSpecs
+        .map(p => {
+          PodSpec.isValid(p) match {
+            case Nil => None
+            case err => Some(PodInvalid(p.id, err))
+          }
+        })
+        .collect { case Some(p) => p }
```

Comment:
just use:
```suggestion
        .flatten
```

Reply:
nice! And in the end, all this can be a flatMap :)